packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -184,6 +184,13 @@ abstract class GlobalKey<T extends State<StatefulWidget>> extends Key {
           ),
         ]);
       }
+      // The parent might have multiple global key reservations. This can happen
+      // when a dirty node under LayoutBuilder is rebuilt before the LayoutBuilder
+      // updates its widget subtree in rendering layer. At the end, the result
+      // of dirty node rebuild will be overwritten by the LayoutBuilder rebuild.
+      // It should be safe to remove all the other reservations of GlobalKeys.
+      // See https://github.com/flutter/flutter/issues/43780
+      _debugReservations.removeWhere((_, Element e) => e == parent);
```

Comment:
Suppose I have erroneous code where parent A has children with keys X and Y, and I write code that erroneously reuses key X for parent B. As I understand it, this code will remove X from the reservation table when Y is processed, thus preventing me from seeing an exception when I then put X under B.

Am I misunderstanding?

Reply:
Yes, you are right. This is not going to work. I will probably have to come up a better solution